test_utils/fake_enterprise_api.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""
+Mock Enterprise API for testing.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from hashlib import md5
+
+import responses
+from faker import Factory as FakerFactory
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.reverse import reverse
+
+from django.conf import settings
+from django.core.cache import cache
+
+
+class EnterpriseMockMixin(object):
+    """
+    Mocks for the Enterprise service response.
+    """
+
+    def setUp(self):
+        """
+        DRY method for EnterpriseMockMixin.
+        """
+        super(EnterpriseMockMixin, self).setUp()
+        cache.clear()
+
+    def build_enterprise_customer_courses_url(self, enterprise_uuid, limit=None, offset=None):
+        """
+        DRY method to make enterprise customer courses url.
+
+        Example URL: 'enterprise/api/v1/enterprise-customer/{enterprise_uuid}/courses'
+        """
+        if limit and offset:
+            enterprise_customer_courses_url = '{lms_root_url}{enterprise_api_uri}?limit={limit}&offset={offset}'.format(
+                lms_root_url=settings.LMS_ROOT_URL,
+                enterprise_api_uri=reverse('enterprise-customer-courses', args=[enterprise_uuid]),
+                limit=limit,
+                offset=offset,
+            )
+        else:
+            enterprise_customer_courses_url = settings.LMS_ROOT_URL + reverse(
+                'enterprise-customer-courses', args=[enterprise_uuid]
+            )
+
+        return enterprise_customer_courses_url
+
+    def build_fake_enterprise_course_detail(self, enterprise_uuid, course_run_key):
+        """
+        DRY method to create course detail json.
+        """
+        course_detail = {
+            'uuid': FakerFactory.create().uuid4(),  # pylint: disable=no-member
+            'key': course_run_key.to_deprecated_string(),
+            'title': 'edX Demonstration Course',
+            'short_description': None,
+            'full_description': None,
+            'start': '2013-02-05T05:00:00Z',
+            'end': None,
+            'enrollment_start': None,
+            'enrollment_end': None,
+            'image': None,
+            'video': None,
+            'min_effort': 5,
+            'max_effort': 3,
+            'course': '{org}+{course}'.format(org=course_run_key.org, course=course_run_key.course),
+            'seats': [
+                {
+                    'sku': self.generate_dummy_sku(course_run_key),
+                    'credit_hours': None,
+                    'price': '100.00',
+                    'currency': 'USD',
+                    'upgrade_deadline': None,
+                    'credit_provider': None,
+                    'type': 'verified'
+                }
+            ],
+            'enrollment_url': reverse(
```

Comment:
This reminds me - we need to make a change [here](https://github.com/edx/edx-enterprise/blob/master/integrated_channels/sap_success_factors/utils.py#L257) so that the transform function uses the result returned in the api instead of constructing the url itself.